src/module/entities/helper.js

```diff
@@ -180,6 +182,83 @@ module.exports = class UserProjectsHelper {
 		})
 	}
 
+	/**
+	 * Fetches targeted roles based on the provided entity IDs.
+	 * @param {Array<string>} entityId - An array of entity IDs to filter roles.
+	 * @param {string} userToken - The token for user authentication.
+	 * @returns {Promise<Object>} A promise that resolves to the response containing the fetched roles or an error object.
+	 */
+	static targetedRoles(entityId, userToken) {
+		return new Promise(async (resolve, reject) => {
+			try {
+				// Construct the filter to retrieve entities based on provided entity IDs
+				const filter = {
+					_id: {
+						$in: entityId,
+					},
+				}
+				const projectionFields = ['childHierarchyPath']
+				// Retrieve entityDetails based on provided entity IDs
+				const entityDetails = await entitiesQueries.entityDocuments(filter, projectionFields)
+				// if (!entityDetails[0].childHierarchyPath) {
+				if (
+					!entityDetails ||
+					!entityDetails[0].childHierarchyPath ||
+					entityDetails[0].childHierarchyPath.length < 0
+				) {
+					throw {
+						status: HTTP_STATUS_CODE.not_found.status,
+						message: CONSTANTS.apiResponses.ENTITY_TYPE_NOT_FOUND,
+					}
+				}
+				// Extract child hierarchy paths from the retrieved entities
+				const childHierarchyPaths = entityDetails[0].childHierarchyPath
+
+				// Construct the filter to retrieve entity type IDs based on child hierarchy paths
+				const entityTypeFilter = {
+					name: {
+						$in: childHierarchyPaths,
+					},
+					isDeleted: false,
+				}
+				const entityTypeProjection = ['_id']
+				// Retrieve entity type IDs based on child hierarchy paths
+				const fetchEntityTypeId = await entityTypeQueries.entityTypesDocument(
+					entityTypeFilter,
+					entityTypeProjection
+				)
+
+				// Check if entity type IDs are retrieved successfully
+				if (!fetchEntityTypeId[0]._id) {
+					throw {
+						status: HTTP_STATUS_CODE.not_found.status,
+						message: CONSTANTS.apiResponses.ENTITY_TYPE_DETAILS_NOT_FOUND,
+					}
+				}
+				// Extract the _id fields from the fetched entity types to use as a filter for user roles
+				const userRoleFilter = fetchEntityTypeId.map((entityType) => entityType._id)
+
+				// Retrieve user roles based on the filtered entity type IDs
+				const userRolesResponse = await userService.userRole(userRoleFilter)
+
+				// Check if user roles are retrieved successfully
+				if (userRolesResponse.data.length < 0) {
```

Comment:
@MallanagoudaB It will be better to add one more check for success=false, hence you are setting it for most of the error situations